prelude/src/stdlib/other/Posix.sk

```diff
@@ -277,6 +278,13 @@ mutable class Popen(
     };
     new_env = env.items().map(e -> `${e.i0}=${e.i1}`).collect(Array);
 
+    // NOTE: Since `posix_spawn()` does not provide a way for changing the
+    // working directory of the child process, we rely on GNU env.
+    cwd match {
+    | Some(path) -> !args = Array["env", "-C", path].concat(args)
+    | None() -> void
+    };
+
```

Comment:
We can't use `posix_spawn_file_actions_addchdir` I guess? This makes me think it would be better to give up on posix_spawn and just use (v)fork/exec so that we can `chdir` and whatever else is needed in the child between `fork` and `exec`. Wrapping the whole thing in `env` is just a heavy-weight version of that right?

Reply:
Yeah at this stage `posix_spawn_file_actions_addchdir_np` is a non-standard extension. I agree that `posix_spawn` is a bit too limiting for our use, and that we should move to `(v)fork`/`exec`. I'll open an issue.